src/event_parser/event_parser.y

```diff
@@ -46,21 +48,20 @@ EventConfMode mode_from_str(const std::string &str) {
   for (const char &c : str) {
     if (m_str.find(c) != std::string::npos) {
       mode |= EventConfMode::kMetric;
-    }
-    if (g_str.find(c) != std::string::npos) {
+    } else if (g_str.find(c) != std::string::npos) {
       mode |= EventConfMode::kCallgraph;
-    }
-    if (l_str.find(c) != std::string::npos) {
+    } else if (l_str.find(c) != std::string::npos) {
         mode |= EventConfMode::kLiveCallgraph;
-    }
-    if (a_str.find(c) != std::string::npos) {
+    } else if (a_str.find(c) != std::string::npos) {
       mode |= EventConfMode::kAll;
+    } else {
+      return {};
```

Comment:
Yeah, this is probably the way to go.

I had originally made this permissive because I wanted to keep ddprof from throwing an error if the user specified a mode which is not compatible with that version of the profiler, but after seeing a few people getting frustrated with the overly-permissive mode specification, I see that was a mistake.

I think we should probably make this more specific in the future, like the user has to specify words instead of just characters.  Like `mode=metric|callgraph` or `mode=m|g`, but I don't think that precise syntax is a good one and I don't think it needs to be done now.